x-pack/legacy/plugins/apm/public/components/shared/TransactionBreakdown/TransactionBreakdownGraph/index.tsx

```diff
@@ -45,7 +57,7 @@ const TransactionBreakdownGraph: React.FC<Props> = props => {
       yMax={1}
       height={unit * 12}
       stacked={true}
-      onHover={trackHoverBreakdownChart}
+      onHover={handleHover}
```

Comment:
I'm having a little bit of trouble tracking the dependencies and what causes what to re-trigger in this flow. The goal is to make sure we only track one hover event every 60s, even if the chart is hovered 100x etc, right?

Could the same thing be achieved with this function defined outside of the component:
```js
const throttler = throttle(fn => fn(), 60000)
```

and onHover would just be 
```js
onHover={() => throttler(() => trackApmEvent({ metric: 'hover_breakdown_chart' }))}
```

Reply:
The problem would be the same as in the initial implementation that the throttling would be reset on every render. We have to somehow memoize the `throttle`d function and for that to work we either do it like what @Kerry350 wrote or we improve `useUiTracker` to `useMemo` the function  it returns.

On second thought improving the `useUiTracker` to return a stable function reference and the `useCallback(() => trackApmEvent({ metric: 'hover_breakdown_chart' }), [trackApmEvent);` might be cleanest (and would help other consumers of the `useUiTracker` too).

Reply:
Yeah - as @weltenwort has explained, a new reference would continually be created, and the throttle reset. I’ll look to improve this on the `useUiTracker` side, that does seem like a better solution.

Reply:
Unless I misunderstand (which is possible), I don't _think_ it would have the same problem in the example I outlined, because the throttling decision is cached and handled outside of the render cycle.

Reply:
Yes, you're right. I didn't read carefully enough and took `throttle(fn => fn())` as pseudo-code :woman_facepalming:

It might be hard to recognize the intention, though, so a stable `useUiTracker` return value and `useMemo`ized `throttle` might be more robust to human misunderstanding and would improve performance for the `onHover` DOM reconciliation at the same time.

I made a mistake in my previous proposal (which I corrected). It should read something like

```typescript
const handleHover = useMemo(() => throttle(
  () => trackApmEvent({ metric: 'hover_breakdown_chart' }),
  60000
), [trackApmEvent]);
```

Reply:
Changed for a A) Stable `useUiTracker` return value and B) A memoized version of the throttled function